internal/collector/collector.go

```diff
@@ -86,6 +99,16 @@ func GetConfig(id string) (Config, error) {
 	return config, nil
 }
 
+// createArchive compresses a directory into a xz-compressed tar archive.
+// Returns an error if the tar command fails.
+func createArchive(archiveName, tmpDir string) error {
+	cmd := exec.Command("tar", "-cJf", archiveName, "-C", tmpDir, ".")
```

Comment:
This command saves the archive into the current directory. Instead, it should store it in a path passed as an additional argument (meaning we'll need to pass two directory values: one to store the data in, other containing the data to compress).

In production, the data store directory will likely be `/var/tmp/rhc/` (to mimic whatever insights-client is doing).